File: history_frontend/src/api/image.ts
import axios from './axios';
import config from '../config';

export interface IImage {
  id: string;
  width: number;
  height: number;
  createdAt: Date;
  url: string;
  bytes: number;
}

const ENDPOINT = '/image/';

export const EDITOR_ENDPOINT = config.apiLink + ENDPOINT + 'editor';

async function upload(file) {
  const formData = new FormData();
  formData.append('image', file);

  const axiosConfig = {
    headers: {
      'content-type': 'multipart/form-data',
    },
  };
  const res = await axios.post<IImage>(
    config.apiLink + ENDPOINT,
    formData,
    axiosConfig
  );
  return res.data;
}

async function remove(id: string) {
  const res = await axios.delete<string>(config.apiLink + ENDPOINT + id);
  return res.data;
}

const exported = {
  upload,
  remove,
};

export default exported;

File: history_backend/history_backend.Domain/DTO/Image.cs
using System;

namespace history_backend.Domain.DTO
{
    public class Image
    {
        public string Id { get; set; }

        public DateTime CreatedAt { get; set; }

        public string Url { get; set; }

        public int Width { get; set; }

        public int Height { get; set; }

        public long Bytes { get; set; }
    }
}
